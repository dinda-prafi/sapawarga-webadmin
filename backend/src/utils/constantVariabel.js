export const ResponseRequest = Object.freeze({
  TIMEOUT: 'ECONNABORTED',
  NETWORKERROR: 'Network Error'
})

export const SurveyStatus = Object.freeze({
  DRAFT: 0,
  ACTIVE: 10
})

export const PopupCategory = Object.freeze({
  EXTERNAL: 'external',
  INTERNAL: 'internal'
})

export const PopupFeature = Object.freeze({
  SURVEY: 'survey',
  POLLING: 'polling',
  NEWS: 'news'
})

export const RolesUser = Object.freeze({
  ADMIN: 'admin',
  STAFFPROV: 'staffProv',
  STAFFKABKOTA: 'staffKabkota',
  STAFFKEC: 'staffKec',
  STAFFKEL: 'staffKel',
  STAFFRW: 'staffRW',
  STAFFSABERHOAX: 'staffSaberhoax',
  TRAINER: 'trainer',
  USER: 'user'
})

export const rolesWidget = {
  ALL: 'all',
  'staffProv': ['admin', 'staffProv'],
  'staffKabkota': ['admin', 'staffProv', 'staffKabkota'],
  'staffKec': ['admin', 'staffProv', 'staffKabkota', 'staffKec'],
  'staffKel': ['admin', 'staffProv', 'staffKabkota', 'staffKec', 'staffKel'],
  'staffRW': ['admin', 'staffProv', 'staffKabkota', 'staffKec', 'staffKel', 'staffRW'],
  'staffSaberhoax': ['admin', 'staffSaberhoax'],
  'trainer': ['admin', 'staffProv', 'trainer'],
  'user': ['admin', 'staffProv', 'user']
}
