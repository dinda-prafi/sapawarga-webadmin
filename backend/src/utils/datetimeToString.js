import moment from 'moment'

export function parsingDatetime(datetime) {
  datetime = moment(datetime * 1000).format('DD MMMM YYYY HH:mm')
  return datetime
}

export function formatDatetime(datetime, format) {
  datetime = moment(datetime).format(format)
  return datetime
}
