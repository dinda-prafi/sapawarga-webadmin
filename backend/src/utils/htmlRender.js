export function render(html) {
  var tmp = document.createElement('DIV')
  tmp.innerHTML = html
  return tmp.textContent || tmp.innerText || ''
}

export function validUrl(str) {
  var pattern = new RegExp('^(https?:\\/\\/)?' + // protocol
      '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|' + // domain name
      '((\\d{1,3}\\.){3}\\d{1,3}))' + // OR ip (v4) address
      '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*' + // port and path
      '(\\?[;&a-z\\d%_.~+=-]*)?' + // query string
      '(\\#[-a-z\\d_]*)?$', 'i') // fragment locator
    return !!pattern.test(str)
}
