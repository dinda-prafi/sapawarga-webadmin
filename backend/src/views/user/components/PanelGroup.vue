<template>
  <div class="app-container">
    <el-row :gutter="40" class="panel-group">
      <el-card class="box-card">
        <div slot="header" class="clearfix">
          <span>{{ $t('label.count-user-each-role') }}:</span>
        </div>
        <div style="margin-bottom:50px;">
          <el-col
            v-if="(pages == rolesWidget.ALL)"
            :xs="12"
            :sm="12"
            :lg="6"
            class="card-panel-col total-all"
          >
            <div class="card-panel">
              <div class="card-panel-icon-wrapper icon-people-all">
                <svg-icon icon-class="peoples" class-name="card-panel-icon" />
              </div>
              <div class="card-panel-description">
                <div class="card-panel-text">
                  {{ $t('label.widget-all-user') }}
                </div>
                <count-to :start-val="0" :end-val="totalAllUser" :duration="duration" class="card-panel-num" />
              </div>
            </div>
          </el-col>
          <el-col
            v-if="(rolesWidget[RolesUser.STAFFPROV].includes(roleId) && (pages != rolesWidget.ALL || roleId != RolesUser.STAFFPROV))"
            :xs="12"
            :sm="12"
            :lg="6"
            class="card-panel-col total-province"
          >
            <div class="card-panel">
              <div class="card-panel-icon-wrapper icon-people-province">
                <svg-icon icon-class="peoples" class-name="card-panel-icon" />
              </div>
              <div class="card-panel-description">
                <div class="card-panel-text">
                  {{ $t('label.widget-province') }}
                </div>
                <count-to :start-val="0" :end-val="totalUserProvince" :duration="duration" class="card-panel-num" />
              </div>
            </div>
          </el-col>
          <el-col
            v-if="(rolesWidget[RolesUser.STAFFKABKOTA].includes(roleId) && (pages != rolesWidget.ALL || roleId != RolesUser.STAFFKABKOTA))"
            :xs="12"
            :sm="12"
            :lg="6"
            class="card-panel-col total-kota"
          >
            <div class="card-panel">
              <div class="card-panel-icon-wrapper icon-people-kota">
                <svg-icon icon-class="peoples" class-name="card-panel-icon" />
              </div>
              <div class="card-panel-description">
                <div class="card-panel-text">
                  {{ $t('label.widget-kabkota') }}
                </div>
                <count-to :start-val="0" :end-val="totalUserKabKota" :duration="duration" class="card-panel-num" />
              </div>
            </div>
          </el-col>
          <el-col
            v-if="(rolesWidget[RolesUser.STAFFKEC].includes(roleId) && (pages != rolesWidget.ALL || roleId != RolesUser.STAFFKEC))"
            :xs="12"
            :sm="12"
            :lg="6"
            class="card-panel-col total-kec"
          >
            <div class="card-panel">
              <div class="card-panel-icon-wrapper icon-people-kec">
                <svg-icon icon-class="peoples" class-name="card-panel-icon" />
              </div>
              <div class="card-panel-description">
                <div class="card-panel-text">
                  {{ $t('label.widget-kecamatan') }}
                </div>
                <count-to :start-val="0" :end-val="totalUserKec" :duration="duration" class="card-panel-num" />
              </div>
            </div>
          </el-col>
          <el-col
            v-if="(rolesWidget[RolesUser.STAFFKEL].includes(roleId) && (pages != rolesWidget.ALL || roleId != RolesUser.STAFFKEL))"
            :xs="12"
            :sm="12"
            :lg="6"
            class="card-panel-col total-kel"
          >
            <div class="card-panel">
              <div class="card-panel-icon-wrapper icon-people-kel">
                <svg-icon icon-class="peoples" class-name="card-panel-icon" />
              </div>
              <div class="card-panel-description">
                <div class="card-panel-text">
                  {{ $t('label.widget-kelurahan') }}
                </div>
                <count-to :start-val="0" :end-val="totalUserKel" :duration="duration" class="card-panel-num" />
              </div>
            </div>
          </el-col>
          <el-col
            v-if="(rolesWidget[RolesUser.STAFFRW].includes(roleId))"
            :xs="12"
            :sm="12"
            :lg="6"
            class="card-panel-col total-rw"
          >
            <div class="card-panel">
              <div class="card-panel-icon-wrapper icon-people-rw">
                <svg-icon icon-class="peoples" class-name="card-panel-icon" />
              </div>
              <div class="card-panel-description">
                <div class="card-panel-text">
                  {{ $t('label.widget-rw') }}
                </div>
                <count-to :start-val="0" :end-val="totalUserRw" :duration="duration" class="card-panel-num" />
              </div>
            </div>
          </el-col>
          <el-col
            v-if="(rolesWidget[RolesUser.STAFFSABERHOAX].includes(roleId))"
            :xs="12"
            :sm="12"
            :lg="6"
            class="card-panel-col total-saber-hoax"
          >
            <div class="card-panel">
              <div class="card-panel-icon-wrapper icon-people-saber-hoax">
                <svg-icon icon-class="peoples" class-name="card-panel-icon" />
              </div>
              <div class="card-panel-description">
                <div class="card-panel-text">
                  {{ $t('label.widget-saber-hoax') }}
                </div>
                <count-to :start-val="0" :end-val="totalUserSaberHoax" :duration="duration" class="card-panel-num" />
              </div>
            </div>
          </el-col>
          <el-col
            v-if="(rolesWidget[RolesUser.TRAINER].includes(roleId))"
            :xs="12"
            :sm="12"
            :lg="6"
            class="card-panel-col total-trainer"
          >
            <div class="card-panel">
              <div class="card-panel-icon-wrapper icon-people-trainer">
                <svg-icon icon-class="peoples" class-name="card-panel-icon" />
              </div>
              <div class="card-panel-description">
                <div class="card-panel-text">
                  {{ $t('label.widget-trainer') }}
                </div>
                <count-to :start-val="0" :end-val="totalUserTrainer" :duration="duration" class="card-panel-num" />
              </div>
            </div>
          </el-col>
          <el-col
            v-if="(rolesWidget[RolesUser.PUBLIK].includes(roleId))"
            :xs="12"
            :sm="12"
            :lg="6"
            class="card-panel-col total-user"
          >
            <div class="card-panel">
              <div class="card-panel-icon-wrapper icon-people-user">
                <svg-icon icon-class="peoples" class-name="card-panel-icon" />
              </div>
              <div class="card-panel-description">
                <div class="card-panel-text">
                  {{ $t('label.widget-user') }}
                </div>
                <count-to :start-val="0" :end-val="totalUserPublik" :duration="duration" class="card-panel-num" />
              </div>
            </div>
          </el-col>
        </div>
      </el-card>
    </el-row>
  </div>
</template>

<script>
import CountTo from 'vue-count-to'
import { RolesUser, rolesWidget } from '@/utils/constantVariabel'

export default {
  name: 'PanelGroupUser',
  components: {
    CountTo
  },
  props: {
    roleId: {
      type: String,
      default: null
    },
    pages: {
      type: String,
      default: null
    },
    duration: {
      type: Number,
      default: 2600
    },
    totalAllUser: {
      type: Number,
      default: 0
    },
    totalUserProvince: {
      type: Number,
      default: 0
    },
    totalUserKabKota: {
      type: Number,
      default: 0
    },
    totalUserKec: {
      type: Number,
      default: 0
    },
    totalUserKel: {
      type: Number,
      default: 0
    },
    totalUserRw: {
      type: Number,
      default: 0
    },
    totalUserTrainer: {
      type: Number,
      default: 0
    },
    totalUserSaberHoax: {
      type: Number,
      default: 0
    },
    totalUserPublik: {
      type: Number,
      default: 0
    }
  },
  data() {
    return {
      RolesUser,
      rolesWidget
    }
  }
}
</script>

<style lang="scss" scoped>
.panel-group {
  margin-top: 18px;
  .card-panel-col{
    margin-bottom: 32px;
  }
  .card-panel {
    height: 108px;
    cursor: pointer;
    font-size: 12px;
    position: relative;
    overflow: hidden;
    color: #666;
    background: #fff;
    box-shadow: 4px 4px 40px rgba(0, 0, 0, .05);
    border-color: rgba(0, 0, 0, .05);
    &:hover {
      .card-panel-icon-wrapper {
        color: #fff;
      }
      .icon-people-all {
         background: #40c9c6;
      }
      .icon-people-province {
         background: #4048c9;
      }
      .icon-people-kota {
         background: #c94096;
      }
      .icon-people-kec {
         background: #6cc940;
      }
      .icon-people-kel {
         background: #c8da49;
      }
      .icon-people-rw {
         background: #dabd49;
      }
      .icon-people-rt {
         background: #76888e;
      }
      .icon-message {
        background: #36a3f7;
      }
      .icon-money {
        background: #f4516c;
      }
      .icon-shopping {
        background: #34bfa3
      }
    }
    .icon-people-all {
      color: #40c9c6;
    }
    .icon-people-province {
      color: #4048c9;
    }
    .icon-people-kota {
      color: #c94096;
    }
    .icon-people-kec {
      color: #6cc940;
    }
    .icon-people-kel {
      color: #c8da49;
    }
    .icon-people-rw {
      color: #dabd49;
    }
    .icon-people-rt {
      color: #76888e;
    }
    .icon-people-trainer {
      color: #7a4880;
    }
    .icon-people-user {
      color: #4eaaff;
    }
    .icon-message {
      color: #36a3f7;
    }
    .icon-money {
      color: #f4516c;
    }
    .icon-shopping {
      color: #34bfa3
    }
    .card-panel-icon-wrapper {
      float: left;
      margin: 14px 0 0 14px;
      padding: 16px;
      transition: all 0.38s ease-out;
      border-radius: 6px;
    }
    .card-panel-icon {
      float: left;
      font-size: 48px;
    }
    .card-panel-description {
      float: right;
      font-weight: bold;
      margin: 20px;
      margin-left: 0px;
      .card-panel-text {
        line-height: 18px;
        color: rgba(0, 0, 0, 0.45);
        font-size: 16px;
        margin-bottom: 12px;
      }
      .card-panel-text-jsh {
        line-height: 18px;
        color: rgba(0, 0, 0, 0.45);
        font-size: 16px;
        margin-bottom: 12px;
      }
      .card-panel-num {
        font-size: 20px;
      }
    }
  }
}
</style>
