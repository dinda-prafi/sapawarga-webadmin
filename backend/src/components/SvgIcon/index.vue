<template>
  <svg :class="svgClass" aria-hidden="true" v-on="$listeners">
    <use :xlink:href="iconName" />
  </svg>
</template>

<script>
import { RolesUser } from '@/utils/constantVariable'
import store from '@/store'

export default {
  name: 'SvgIcon',
  props: {
    iconClass: {
      type: String,
      required: true
    },
    className: {
      type: String,
      default: ''
    }
  },
  computed: {
    iconName() {
      return `#icon-${this.iconClass}`
    },
    svgClass() {
      console.log()
      if (this.className) {
        return 'svg-icon ' + this.className
      } else if (store.getters.roles[0] === RolesUser.LEADER) {
        return 'svg-icon leader-icon-menu'
      } else {
        return 'svg-icon'
      }
    }
  }
}
</script>

<style scoped>
.svg-icon {
  width: 1em;
  height: 1em;
  vertical-align: -0.15em;
  fill: currentColor;
  overflow: hidden;
}
</style>
