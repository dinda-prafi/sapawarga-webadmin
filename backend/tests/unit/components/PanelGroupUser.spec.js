import { mount, createLocalVue } from "@vue/test-utils";
import PanelGroup from "@/views/user/components/PanelGroup";
import totalUser from "../fixtures/totalUser";
import ElementUI from "element-ui";
import { RolesUser, rolesWidget } from '@/utils/constantVariabel'
//import SVG from '@/components/SvgIcon';
// src/components/SvgIcon/index.vue
import SvgIcon from "@/components/SvgIcon"; // svg组件
import i18n from '@/lang'

// register globally

const localVue = createLocalVue();
localVue.use(ElementUI);
localVue.component("svg-icon", SvgIcon);

const totalAllUser = totalUser.data.items[0].value;
const totalUserProvince = totalUser.data.items[1].value;
const totalUserKabKota = totalUser.data.items[2].value;
const totalUserKec = totalUser.data.items[3].value;
const totalUserKel = totalUser.data.items[4].value;
const totalUserRw = totalUser.data.items[5].value;
const totalUserSaberHoax = totalUser.data.items[6].value;
const totalUserTrainer = totalUser.data.items[7].value;
const totalUserPengguna = totalUser.data.items[8].value;

describe("PanelGroup User", () => {
  let props;
  const selectorAll = ".total-all";
  const selectorUserProvince = ".total-province";
  const selectorUserKabKota = ".total-kota";
  const selectorUserKec = ".total-kec";
  const selectorUserKel = ".total-kel";
  const selectorUserRw = ".total-rw";
  const selectorUserSaberHoax = ".total-saber-hoax";
  const selectorUserTrainer = ".total-trainer";
  const selectorUserPengguna = ".total-user";

  const build = () => {
    const wrapper = mount(PanelGroup, {
      RolesUser,
      rolesWidget,
      propsData: props,
      localVue,
      i18n
    });
    return {
      wrapper,
      cardColumnAllUser: () => wrapper.find(".card-panel-col" + selectorAll),
      cardIconAllUser: () => wrapper.find(selectorAll).find(".card-panel-icon-wrapper"),
      cardTextAllUser: () => wrapper.find(selectorAll).find(".card-panel-text"),
      cardNumberAllUser: () => wrapper.find(selectorAll).find(".card-panel-num"),
      cardColumnUserProvince: () =>
        wrapper.find(".card-panel-col" + selectorUserProvince),
      cardIconUserProvince: () =>
        wrapper.find(selectorUserProvince).find(".card-panel-icon-wrapper"),
      cardTextUserProvince: () =>
        wrapper.find(selectorUserProvince).find(".card-panel-text"),
      cardNumberUserProvince: () =>
        wrapper.find(selectorUserProvince).find(".card-panel-num"),
      cardColumnUserKabKota: () =>
        wrapper.find(".card-panel-col" + selectorUserKabKota),
      cardIconUserKabKota: () =>
        wrapper.find(selectorUserKabKota).find(".card-panel-icon-wrapper"),
      cardTextUserKabKota: () =>
        wrapper.find(selectorUserKabKota).find(".card-panel-text"),
      cardNumberUserKabKota: () =>
        wrapper.find(selectorUserKabKota).find(".card-panel-num"),
      cardColumnUserKec: () =>
        wrapper.find(".card-panel-col" + selectorUserKec),
      cardIconUserKec: () =>
        wrapper.find(selectorUserKec).find(".card-panel-icon-wrapper"),
      cardTextUserKec: () =>
        wrapper.find(selectorUserKec).find(".card-panel-text"),
      cardNumberUserKec: () =>
        wrapper.find(selectorUserKec).find(".card-panel-num"),
      cardColumnUserKel: () =>
        wrapper.find(".card-panel-col" + selectorUserKel),
      cardIconUserKel: () =>
        wrapper.find(selectorUserKel).find(".card-panel-icon-wrapper"),
      cardTextUserKel: () =>
        wrapper.find(selectorUserKel).find(".card-panel-text"),
      cardNumberUserKel: () =>
        wrapper.find(selectorUserKel).find(".card-panel-num"),
      cardColumnUserRw: () => wrapper.find(".card-panel-col" + selectorUserRw),
      cardIconUserRw: () =>
        wrapper.find(selectorUserRw).find(".card-panel-icon-wrapper"),
      cardTextUserRw: () =>
        wrapper.find(selectorUserRw).find(".card-panel-text"),
      cardNumberUserRw: () =>
        wrapper.find(selectorUserRw).find(".card-panel-num"),
      cardColumnUserSaberHoax: () => wrapper.find(".card-panel-col" + selectorUserSaberHoax),
      cardIconUserSaberHoax: () =>
        wrapper.find(selectorUserSaberHoax).find(".card-panel-icon-wrapper"),
      cardTextUserSaberHoax: () =>
        wrapper.find(selectorUserSaberHoax).find(".card-panel-text"),
      cardNumberUserSaberHoax: () =>
        wrapper.find(selectorUserSaberHoax).find(".card-panel-num"),
      cardColumnUserTrainer: () => wrapper.find(".card-panel-col" + selectorUserTrainer),
      cardIconUserTrainer: () =>
        wrapper.find(selectorUserTrainer).find(".card-panel-icon-wrapper"),
      cardTextUserTrainer: () =>
        wrapper.find(selectorUserTrainer).find(".card-panel-text"),
      cardNumberUserTrainer: () =>
        wrapper.find(selectorUserTrainer).find(".card-panel-num"),
      cardColumnUserPengguna: () => wrapper.find(".card-panel-col" + selectorUserPengguna),
      cardIconUserPengguna: () =>
        wrapper.find(selectorUserPengguna).find(".card-panel-icon-wrapper"),
      cardTextUserPengguna: () =>
        wrapper.find(selectorUserPengguna).find(".card-panel-text"),
      cardNumberUserPengguna: () =>
        wrapper.find(selectorUserPengguna).find(".card-panel-num")
    };
  };

  beforeEach(() => {
    props = {
      roleId: 'admin',
      pages: 'all',
      rolesWidget: RolesUser,
      RolesUser: rolesWidget,
      totalAllUser: totalAllUser,
      totalUserProvince: totalUserProvince,
      totalUserKabKota: totalUserKabKota,
      totalUserKec: totalUserKec,
      totalUserKel: totalUserKel,
      totalUserRw: totalUserRw,
      totalUserSaberHoax: totalUserSaberHoax,
      totalUserTrainer: totalUserTrainer,
      totalUserPengguna: totalUserPengguna,
      duration: 1
    };
  });

  it("renders the component", () => {
    // arrange
    const { wrapper } = build();

    // assert
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders all card", done => {
    // arrange
    const {
      wrapper,
      cardColumnAllUser,
      cardIconAllUser,
      cardTextAllUser,
      cardNumberAllUser,
      cardColumnUserProvince,
      cardIconUserProvince,
      cardTextUserProvince,
      cardNumberUserProvince,
      cardColumnUserKabKota,
      cardIconUserKabKota,
      cardTextUserKabKota,
      cardNumberUserKabKota,
      cardColumnUserKec,
      cardIconUserKec,
      cardTextUserKec,
      cardNumberUserKec,
      cardColumnUserKel,
      cardIconUserKel,
      cardTextUserKel,
      cardNumberUserKel,
      cardColumnUserRw,
      cardIconUserRw,
      cardTextUserRw,
      cardNumberUserRw,
      cardColumnUserSaberHoax,
      cardIconUserSaberHoax,
      cardTextUserSaberHoax,
      cardNumberUserSaberHoax,
      cardColumnUserTrainer,
      cardIconUserTrainer,
      cardTextUserTrainer,
      cardNumberUserTrainer,
      cardColumnUserPengguna,
      cardIconUserPengguna,
      cardTextUserPengguna,
      cardNumberUserPengguna
    } = build();

    // assert
    expect(wrapper.props().roleId).toBe("admin");
    expect(cardColumnAllUser().exists()).toBe(true);
    expect(cardIconAllUser().exists()).toBe(true);
    expect(cardTextAllUser().exists()).toBe(true);
    expect(cardNumberAllUser().exists()).toBe(true);
    // expect(wrapper.vm.roleId).toBeTruthy();
    expect(cardColumnUserProvince().exists()).toBe(true);
    expect(cardIconUserProvince().exists()).toBe(true);
    expect(cardTextUserProvince().exists()).toBe(true);
    expect(cardNumberUserProvince().exists()).toBe(true);
    expect(cardColumnUserKabKota().exists()).toBe(true);
    expect(cardIconUserKabKota().exists()).toBe(true);
    expect(cardTextUserKabKota().exists()).toBe(true);
    expect(cardNumberUserKabKota().exists()).toBe(true);
    expect(cardColumnUserKec().exists()).toBe(true);
    expect(cardIconUserKec().exists()).toBe(true);
    expect(cardTextUserKec().exists()).toBe(true);
    expect(cardNumberUserKec().exists()).toBe(true);
    expect(cardColumnUserKel().exists()).toBe(true);
    expect(cardIconUserKel().exists()).toBe(true);
    expect(cardTextUserKel().exists()).toBe(true);
    expect(cardNumberUserKel().exists()).toBe(true);
    expect(cardColumnUserRw().exists()).toBe(true);
    expect(cardIconUserRw().exists()).toBe(true);
    expect(cardTextUserRw().exists()).toBe(true);
    expect(cardNumberUserRw().exists()).toBe(true);
    expect(cardColumnUserSaberHoax().exists()).toBe(true);
    expect(cardIconUserSaberHoax().exists()).toBe(true);
    expect(cardTextUserSaberHoax().exists()).toBe(true);
    expect(cardNumberUserSaberHoax().exists()).toBe(true);
    expect(cardColumnUserTrainer().exists()).toBe(true);
    expect(cardIconUserTrainer().exists()).toBe(true);
    expect(cardTextUserTrainer().exists()).toBe(true);
    expect(cardNumberUserTrainer().exists()).toBe(true);
    expect(cardNumberAllUser().text()).toMatch("0");
    expect(cardTextAllUser().text()).toMatch("Semua");
    expect(cardNumberUserProvince().text()).toMatch("0");
    expect(cardTextUserProvince().text()).toMatch("Provinsi");
    expect(cardNumberUserKabKota().text()).toMatch("0");
    expect(cardTextUserKabKota().text()).toMatch("Kota/Kab");
    expect(cardNumberUserKec().text()).toMatch("0");
    expect(cardTextUserKec().text()).toMatch("Kecamatan");
    expect(cardNumberUserKel().text()).toMatch("0");
    expect(cardTextUserKel().text()).toMatch("Desa/Kel");
    expect(cardNumberUserRw().text()).toMatch("0");
    expect(cardTextUserRw().text()).toMatch("RW");
    expect(cardTextUserSaberHoax().text()).toMatch("Saber Hoax");
    expect(cardNumberUserSaberHoax().text()).toMatch("0");
    expect(cardTextUserSaberHoax().text()).toMatch("Saber Hoax");
    expect(cardTextUserTrainer().text()).toMatch("Pelatih");
    expect(cardNumberUserTrainer().text()).toMatch("0");
    expect(cardTextUserTrainer().text()).toMatch("Pelatih");
    expect(cardTextUserPengguna().text()).toMatch("Pengguna");
    expect(cardNumberUserPengguna().text()).toMatch("0");
    expect(cardTextUserPengguna().text()).toMatch("Pengguna");
    setTimeout(() => {
      expect(cardNumberAllUser().text()).toMatch(totalAllUser.toString());
      expect(cardNumberUserProvince().text()).toMatch(
        totalUserProvince.toString()
      );
      expect(cardNumberUserKabKota().text()).toMatch(
        totalUserKabKota.toString()
      );
      expect(cardNumberUserKec().text()).toMatch(totalUserKec.toString());
      expect(cardNumberUserKel().text()).toMatch(totalUserKel.toString());
      expect(cardNumberUserRw().text()).toMatch(totalUserRw.toString());
      expect(cardNumberUserSaberHoax().text()).toMatch(totalUserSaberHoax.toString());
      expect(cardNumberUserTrainer().text()).toMatch(totalUserTrainer.toString());
      expect(cardNumberUserPengguna().text()).toMatch(totalUserPengguna.toString());
      done();
    }, 1100);

    expect(wrapper.props().totalAllUser).toBe(totalAllUser);
    expect(wrapper.props().totalUserProvince).toBe(totalUserProvince);
    expect(wrapper.props().totalUserKabKota).toBe(totalUserKabKota);
    expect(wrapper.props().totalUserKec).toBe(totalUserKec);
    expect(wrapper.props().totalUserKel).toBe(totalUserKel);
    expect(wrapper.props().totalUserRw).toBe(totalUserRw);
    expect(wrapper.props().totalUserSaberHoax).toBe(totalUserSaberHoax);
    expect(wrapper.props().totalUserTrainer).toBe(totalUserTrainer);
    expect(wrapper.props().totalUserPengguna).toBe(totalUserPengguna);
    // expect(name().text()).toBe(props.user.name)
  });

  it("hide total all user when role is not admin", () => {
    // arrange
    const { wrapper } = build();
    const mainSelector = ".total-all";
    wrapper.setProps({ roleId: "staffProv" });

    const cardColumn = () => wrapper.find(".card-panel-col" + mainSelector);

    expect(cardColumn().exists()).toBe(true);
  });

  it("hide total all user province when role is not province", () => {
    // arrange
    const { wrapper } = build();
    const mainSelector = ".total-province";
    wrapper.setProps({ roleId: "staffKabkota" });

    const cardColumn = () => wrapper.find(".card-panel-col" + mainSelector);

    expect(cardColumn().exists()).toBe(false);
  });

  it("hide total all user kota/kab when role is not kota/kab", () => {
    // arrange
    const { wrapper } = build();
    const mainSelector = ".total-kota";
    wrapper.setProps({ roleId: "staffKec" });

    const cardColumn = () => wrapper.find(".card-panel-col" + mainSelector);

    expect(cardColumn().exists()).toBe(false);
  });

  it("hide total all user kecamatan when role is not kecamatan", () => {
    // arrange
    const { wrapper } = build();
    const mainSelector = ".total-kec";
    wrapper.setProps({ roleId: "staffKel" });

    const cardColumn = () => wrapper.find(".card-panel-col" + mainSelector);

    expect(cardColumn().exists()).toBe(false);
  });

  it("hide total all user kelurahan when role is not kelurahan", () => {
    // arrange
    const { wrapper } = build();
    const mainSelector = ".total-kel";
    wrapper.setProps({ roleId: "staffRW" });

    const cardColumn = () => wrapper.find(".card-panel-col" + mainSelector);

    expect(cardColumn().exists()).toBe(false);
  });

  it("hide total all user saber hoax when role is not saber hoax", () => {
    // arrange
    const { wrapper } = build();
    const mainSelector = ".total-saber-hoax";
    wrapper.setProps({ roleId: "saberHoax" });

    const cardColumn = () => wrapper.find(".card-panel-col" + mainSelector);

    expect(cardColumn().exists()).toBe(false);
  });

  it("hide total all user trainer when role is not trainer", () => {
    // arrange
    const { wrapper } = build();
    const mainSelector = ".total-trainer";
    wrapper.setProps({ roleId: "trainer" });

    const cardColumn = () => wrapper.find(".card-panel-col" + mainSelector);

    expect(cardColumn().exists()).toBe(true);
  });

  it("hide total all user pengguna when role is not use", () => {
    // arrange
    const { wrapper } = build();
    const mainSelector = ".total-user";
    wrapper.setProps({ roleId: "user" });

    const cardColumn = () => wrapper.find(".card-panel-col" + mainSelector);

    expect(cardColumn().exists()).toBe(true);
  });
});
