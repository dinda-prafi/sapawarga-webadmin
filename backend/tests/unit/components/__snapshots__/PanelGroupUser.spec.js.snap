// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelGroup User renders the component 1`] = `
<div class="app-container">
  <div class="panel-group el-row" style="margin-left: -20px; margin-right: -20px;">
    <div class="el-card box-card is-always-shadow">
      <div class="el-card__header">
        <div class="clearfix"><span>Total user per:</span></div>
      </div>
      <div class="el-card__body">
        <div style="margin-bottom: 50px;">
          <div class="card-panel-col total-all el-col el-col-24 el-col-xs-12 el-col-sm-12 el-col-lg-6" style="padding-left: 20px; padding-right: 20px;">
            <div class="card-panel">
              <div class="card-panel-icon-wrapper icon-people-all"><svg aria-hidden="true" class="svg-icon card-panel-icon">
                  <use xlink:href="#icon-peoples"></use>
                </svg></div>
              <div class="card-panel-description">
                <div class="card-panel-text">
                  Semua
                </div> <span class="card-panel-num">
  0
</span>
              </div>
            </div>
          </div>
          <div class="card-panel-col total-province el-col el-col-24 el-col-xs-12 el-col-sm-12 el-col-lg-6" style="padding-left: 20px; padding-right: 20px;">
            <div class="card-panel">
              <div class="card-panel-icon-wrapper icon-people-province"><svg aria-hidden="true" class="svg-icon card-panel-icon">
                  <use xlink:href="#icon-peoples"></use>
                </svg></div>
              <div class="card-panel-description">
                <div class="card-panel-text">
                  Provinsi
                </div> <span class="card-panel-num">
  0
</span>
              </div>
            </div>
          </div>
          <div class="card-panel-col total-kota el-col el-col-24 el-col-xs-12 el-col-sm-12 el-col-lg-6" style="padding-left: 20px; padding-right: 20px;">
            <div class="card-panel">
              <div class="card-panel-icon-wrapper icon-people-kota"><svg aria-hidden="true" class="svg-icon card-panel-icon">
                  <use xlink:href="#icon-peoples"></use>
                </svg></div>
              <div class="card-panel-description">
                <div class="card-panel-text">
                  Kota/Kab
                </div> <span class="card-panel-num">
  0
</span>
              </div>
            </div>
          </div>
          <div class="card-panel-col total-kec el-col el-col-24 el-col-xs-12 el-col-sm-12 el-col-lg-6" style="padding-left: 20px; padding-right: 20px;">
            <div class="card-panel">
              <div class="card-panel-icon-wrapper icon-people-kec"><svg aria-hidden="true" class="svg-icon card-panel-icon">
                  <use xlink:href="#icon-peoples"></use>
                </svg></div>
              <div class="card-panel-description">
                <div class="card-panel-text">
                  Kecamatan
                </div> <span class="card-panel-num">
  0
</span>
              </div>
            </div>
          </div>
          <div class="card-panel-col total-kel el-col el-col-24 el-col-xs-12 el-col-sm-12 el-col-lg-6" style="padding-left: 20px; padding-right: 20px;">
            <div class="card-panel">
              <div class="card-panel-icon-wrapper icon-people-kel"><svg aria-hidden="true" class="svg-icon card-panel-icon">
                  <use xlink:href="#icon-peoples"></use>
                </svg></div>
              <div class="card-panel-description">
                <div class="card-panel-text">
                  Desa/Kel
                </div> <span class="card-panel-num">
  0
</span>
              </div>
            </div>
          </div>
          <div class="card-panel-col total-rw el-col el-col-24 el-col-xs-12 el-col-sm-12 el-col-lg-6" style="padding-left: 20px; padding-right: 20px;">
            <div class="card-panel">
              <div class="card-panel-icon-wrapper icon-people-rw"><svg aria-hidden="true" class="svg-icon card-panel-icon">
                  <use xlink:href="#icon-peoples"></use>
                </svg></div>
              <div class="card-panel-description">
                <div class="card-panel-text">
                  RW
                </div> <span class="card-panel-num">
  0
</span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
